src/lib/html.ts

```diff
@@ -56,35 +75,60 @@ export function generateHtml(
 }
 
 /**
- * Create a combined result where each of the inputs is a separate node under the root
+ * Create a combined tree data where each of the inputs is a separate node under the root
  */
-function makeMergedBundle(exploreResults: ExploreBundleResult[]): ExploreBundleResult {
-  let totalBytes = 0;
-  const files: FileDataMap = {};
+export function makeMergedTreeDataMap(treeData: WebTreeData[]): WebTreeData {
+  const data: WebTreeMapNode = newNode('/');
 
-  // Remove any common prefix to keep the visualization as simple as possible.
-  const commonPrefix = getCommonPathPrefix(exploreResults.map((r) => r.bundleName));
+  data.children = [];
 
-  for (const result of exploreResults) {
-    totalBytes += result.totalBytes;
+  for (const result of treeData) {
+    const childTree = result.data;
 
-    const prefix = result.bundleName.slice(commonPrefix.length);
+    childTree.name = result.name;
 
-    Object.entries(result.files).forEach(([fileName, size]) => {
-      files[`${prefix}/${fileName}`] = size;
-    });
+    data.data['$area'] += childTree.data['$area'];
+    data.children.push(childTree);
   }
 
+  removeSameRootPaths(data.children);
+
   return {
-    bundleName: '[combined]',
-    totalBytes,
-    mappedBytes: 0,
-    eolBytes: 0,
-    sourceMapCommentBytes: 0,
-    files,
+    name: COMBINED_BUNDLE_NAME,
+    data,
   };
 }
 
+/**
+ * Find and remove the same root paths
+ */
+function removeSameRootPaths(nodes: WebTreeMapNode[]): void {
```

Comment:
I believe `getCommonPathPrefix` can handle this logic

Reply:
Okay, I removed this function.